### ISSUE TITLE
coverallsapp/github-action@v2 seems broken?

### ISSUE THREAD

**Issue by yx-lamini** [coverallsapp/github-action#224]
It's reporting a gz file does not exist?
![image](https://github.com/user-attachments/assets/f5f48823-c725-45ff-a336-2f2ce15eb13d)

The coverall step in CI looks as follows:
![image](https://github.com/user-attachments/assets/6f157e06-a6e7-49fd-adef-313eb36afedd)

**Comment by mfrr1118** [comment]
Same here. Started happening in the last hour or so for us

**Comment by Zeitsperre** [comment]
Also am seeing this now in my projects.

**Comment by oldbonsai** [comment]
I’ll add another voice to the thread. I’ve been having that failure in our CI for about the same time — the last hour or so.

**Comment by patattack69** [comment]
I also have the same issue with our builds

**Comment by cspeper** [comment]
👍

**Comment by afinetooth** [comment]
@yx-lamini Thanks for reporting. We believe this is the underlying issue is coverage-reporter: https://github.com/coverallsapp/coverage-reporter/issues/165

Working on it now. Will update everyone here.

**Comment by afinetooth** [comment]
Suggestion in meantime:

set [fail-on-error input option](https://github.com/marketplace/actions/coveralls-github-action#:~:text=fail%2Don%2Derror,to%20any%20errors.) to `false` for now:

```
fail-on-error: false
```

**Comment by afinetooth** [comment]
@yx-lamini @mfrr1118 @Zeitsperre @oldbonsai @patattack69 @cspeper We believe we've patched this for now by updating the coveralls-checksums.txt file.

Can you please try again to see if it works for you now?

**Comment by zmariscal** [comment]
I have `fail-on-error: false` set and it still fails unfortunately. I also see the original issue has a screenshot with this set as well.

**Comment by afinetooth** [comment]
@zmariscal Thanks. Another user reported the same. We'll have to address that next.

In the meantime, we think we've got it patched. Can you try again and let us know please?

**Comment by Zeitsperre** [comment]
Confirmed working on my end. Thanks so much!

**Comment by afinetooth** [comment]
Thank you @Zeitsperre!

We're now fixing the root cause for all subsequent releases.

**Comment by oldbonsai** [comment]
Also confirmed as working. Thanks for the quick turnaround on this!

**Comment by brantphoto** [comment]
rca? thanks!

**Comment by afinetooth** [comment]
@brantphoto RCA:

We added support for linux on `aarch64` to this and all future releases of [coverage-reporter](https://github.com/coverallsapp/coverage-reporter/releases/tag/v0.6.15) (as of `v0.6.15`). 

The cross-compilation process was so gnarly that we ended up with a completely different directory structure in terms of where intermediate files and artifacts were stored and to which they were uploaded and downloaded. 

This resulted in our checksums (`coveralls-checksums.txt`) pointing to binaries with the wrong path names.

Like this:

```
1559b159a93ae44a649fe7ef8f10d906c146057c4f81acb4f448d441b9ff4dadb3  release/coveralls-linux
47653fa86b8eaae30b16c5d3f37fbeda30d8708153a261df2cdc9cb67e6c48e0  release/coveralls-linux-aarch64
50fe2ee0d5c42bce14533ec7b7dcf24c7be0679126ccfd76e76f7851ffb1fe3b  release/coveralls-linux-aarch64.tar.gz
59b159a93ae44a649fe7ef8f10d906c146057c4f81acb4f448d441b9ff4dadb3  release/coveralls-linux-x86_64
b275cd52577f04237f99ab45055b002a8af7d501c57bf9076d013872fd4ea4c2  release/coveralls-linux-x86_64.tar.gz
b275cd52577f04237f99ab45055b002a8af7d501c57bf9076d013872fd4ea4c2  release/coveralls-linux.tar.gz
3cdf77f405958e607128f141cae8e206b5b51ba4e1c942677da93214f316a0ce  release/coveralls-windows.exe
ea4ad54e6d1fc06269f7d4dd5768a109cb31c7f1e907470f5f986a45c0919cb8  release/coveralls-windows.zip
```

When it should have been this:

```
59b159a93ae44a649fe7ef8f10d906c146057c4f81acb4f448d441b9ff4dadb3  coveralls-linux
47653fa86b8eaae30b16c5d3f37fbeda30d8708153a261df2cdc9cb67e6c48e0  coveralls-linux-aarch64
50fe2ee0d5c42bce14533ec7b7dcf24c7be0679126ccfd76e76f7851ffb1fe3b  coveralls-linux-aarch64.tar.gz
59b159a93ae44a649fe7ef8f10d906c146057c4f81acb4f448d441b9ff4dadb3  coveralls-linux-x86_64
b275cd52577f04237f99ab45055b002a8af7d501c57bf9076d013872fd4ea4c2  coveralls-linux-x86_64.tar.gz
b275cd52577f04237f99ab45055b002a8af7d501c57bf9076d013872fd4ea4c2  coveralls-linux.tar.gz
3cdf77f405958e607128f141cae8e206b5b51ba4e1c942677da93214f316a0ce  coveralls-windows.exe
ea4ad54e6d1fc06269f7d4dd5768a109cb31c7f1e907470f5f986a45c0919cb8  coveralls-windows.zip
```

That's why the patch was fixing our checksums text file.

In the meantime, the fix to the root cause is to replace [this line](https://github.com/coverallsapp/coverage-reporter/blob/master/.github/workflows/build.yml#L224) with this one:

```
(cd release && sha256sum * > ../coveralls-checksums.txt)
```

**Comment by mfrr1118** [comment]
All good on our end. Thanks

**Comment by zmariscal** [comment]
> @zmariscal Thanks. Another user reported the same. We'll have to address that next.

Thanks for the fast turn around for the original issue. Would you like for me to create an issue about the fail on failure not working properly?

**Comment by afinetooth** [comment]
Hi @zmariscal. Thanks for the offer re: the `fail-on-error` setting. You are welcome to do that but, FYI, right now I am tracking that through an [existing issue from our orb integration](https://github.com/coverallsapp/orb/issues/40). (There's even a [potential fix](https://github.com/coverallsapp/orb/issues/40#issue-2206272364) there that looks reasonable.)

We intend to apply the fix to both integrations at the same time. 

I'm going to close this for now, per the original issue.

As I say, feel free to create a new github-action issue, or you could just follow the related [orb issue](https://github.com/coverallsapp/orb/issues/40#issue-2206272364).

**Comment by afinetooth** [comment]
@zmariscal wanted to let you know that we just released [`v2.3.1` of the github-action](https://github.com/coverallsapp/github-action/releases), which extends the behavior of the `fail-on-error` input to any failures from "setup" tasks, including:

- Failure to download the binary;
- Failure to verify the binary; and
- Failure to find the downloaded binary after extraction.

This last error with the `github-action` related to a checksum verification error with the latest version of `coverage-reporter` (`v0.6.15`) (which has been patched since), but the reason `fail-on-error` did not work then was because its behavior only extended to failures of the two possible commands issued to the final binary, `coveralls report` or `coveralls done`.

[`v2.3.1` of the github-action](https://github.com/coverallsapp/github-action/releases) now checks for success in each setup step and, on failure of any step, returns exit code `1` unless `fail-on-error` is set to `false`, in which case it returns exit code `0`.

**Comment by zmariscal** [comment]
@afinetooth thanks for letting me know and your work to fix this 🙇🏼
